### PR TITLE
Minor updates to notebook diff editor

### PR DESCRIPTION
**Pull request by DonJayamanne**
For #208280

### CODE REVIEW

File: src/vs/workbench/contrib/notebook/browser/diff/diffElementViewModel.ts [hunk 1]
```diff
@@ -239,8 +239,6 @@ export abstract class DiffElementCellViewModelBase extends DiffElementViewModelB
 		this.cellFoldingState = modified?.getTextBufferHash() !== original?.getTextBufferHash() ? PropertyFoldingState.Expanded : PropertyFoldingState.Collapsed;
 		this.metadataFoldingState = PropertyFoldingState.Collapsed;
 		this.outputFoldingState = PropertyFoldingState.Collapsed;
-
-		this._register(this.editorEventDispatcher.onDidChangeLayout(e => this._layoutInfoEmitter.fire({ outerWidth: true })));
```
Comment (DonJayamanne) [inline src/vs/workbench/contrib/notebook/browser/diff/diffElementViewModel.ts:243]: This is already in the base class ctor

File: src/vs/workbench/contrib/notebook/browser/diff/notebookDiffEditor.ts [hunk 2]
```diff
@@ -617,11 +617,11 @@ export class NotebookTextDiffEditor extends EditorPane implements INotebookTextD
 		}, 10);
 	}
 
-	private pendingLayouts = new WeakMap<DiffElementCellViewModelBase, IDisposable>();
```
Comment (DonJayamanne) [inline src/vs/workbench/contrib/notebook/browser/diff/notebookDiffEditor.ts:620]: Future proofing for #208280

File: src/vs/workbench/contrib/notebook/browser/diff/unchangedEditorRegions.ts [hunk 3]
```diff
@@ -86,6 +86,15 @@ function createHideUnchangedRegionOptions(configurationService: IConfigurationSe
 	};
 
 	disposables.add(configurationService.onDidChangeConfiguration(e => {
+		if (e.affectsConfiguration('diffEditor.hideUnchangedRegions.minimumLineCount')) {
+			options.minimumLineCount = configurationService.getValue<number>('diffEditor.hideUnchangedRegions.minimumLineCount');
```
Comment (DonJayamanne) [inline src/vs/workbench/contrib/notebook/browser/diff/unchangedEditorRegions.ts:90]: Good change for regular nb diff view and also future proofing for #208280

File: src/vs/workbench/contrib/notebook/common/model/notebookCellTextModel.ts [hunk 4]
```diff
@@ -113,12 +112,7 @@ export class NotebookCellTextModel extends Disposable implements ICell {
 			return this._textBuffer;
 		}
 
-		const builder = new PieceTreeTextBufferBuilder();
-		builder.acceptChunk(this._source);
-		const bufferFactory = builder.finish(true);
-		const { textBuffer, disposable } = bufferFactory.create(model.DefaultEndOfLine.LF);
-		this._textBuffer = textBuffer;
-		this._register(disposable);
```
Comment (DonJayamanne) [inline src/vs/workbench/contrib/notebook/common/model/notebookCellTextModel.ts:121]: @rebornix Replacing this as I need to create text buffers for metadata as well, and figure i might as well change this here too.

**Review by lszomoru — APPROVED** [review]

**Pull request merged**